File: packages/ios-sdk/Sources/GateFlowKit/Identity/IdentityLogic.swift
import Foundation

/// Stateless helper for identity operations.
enum IdentityLogic {
    static let superwallPrefix = "sw_"

    /// Generate a random alias ID for anonymous users.
    static func generateAlias() -> String {
        "\(superwallPrefix)\(UUID().uuidString)"
    }

    /// Generate a random seed for bucket computation.
    static func generateSeed() -> Int {
        Int.random(in: 0...Int.max)
    }

    /// Validate and sanitize a user ID.
    static func sanitize(userId: String) -> String? {
        let trimmed = userId.trimmingCharacters(in: .whitespacesAndNewlines)
        return trimmed.isEmpty ? nil : trimmed
    }

    /// Determine if we need to fetch assignments.
    static func shouldGetAssignments(isLoggedIn: Bool, isFirstAppOpen: Bool) -> Bool {
        // Always fetch on first app open or when not logged in (anonymous).
        isFirstAppOpen || !isLoggedIn
    }
}

File: packages/android-sdk/gateflow/src/main/java/com/gateflow/sdk/identity/IdentityLogic.kt
package com.gateflow.sdk.identity

import java.util.UUID

/** Stateless helper for identity operations. */
object IdentityLogic {
    private const val PREFIX = "gf_"

    /** Generate a random alias ID for anonymous users. */
    fun generateAlias(): String = "$PREFIX${UUID.randomUUID()}"

    /** Generate a random seed for bucket computation. */
    fun generateSeed(): Int = (0..Int.MAX_VALUE).random()

    /** Validate and sanitize a user ID. */
    fun sanitize(userId: String): String? {
        val trimmed = userId.trim()
        return trimmed.takeIf { it.isNotEmpty() }
    }

    /** Determine if we need to fetch assignments. */
    fun shouldGetAssignments(isLoggedIn: Boolean, isFirstAppOpen: Boolean): Boolean =
        isFirstAppOpen || !isLoggedIn
}
